front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -55,6 +63,61 @@ export default function PowerRestrictionSelector() {
           ];
           dispatch(updatePowerRestriction(powerRestrictionRange));
         });
+
+      const { data: pathWithCatenaries } = await getPathFindingByIdCatenaries({
+        pathId: pathFindingID,
+      });
+      const { data: rollingStockWithCurves } = await getRollingStockWithCurves({
+        id: rollingStockID,
+      });
+
+      if (rollingStockWithCurves && pathWithCatenaries) {
+        // Extract uniques rollingstock's power restriction codes allowed by each electrification mode
+        const curvesMode = rollingStockWithCurves?.effort_curves.modes;
+        const curvesModesKey = Object.keys(curvesMode);
+
+        let parsedElectrification: ElectrificationPR = {};
+        for (const mode of curvesModesKey) {
+          let powerCodes = curvesMode[mode].curves.map(
+            (curve) => curve.cond?.power_restriction_code
+          );
+          powerCodes = powerCodes.filter(
+            (value, index, array) => value && array.indexOf(value) === index
+          );
+          parsedElectrification = {
+            ...parsedElectrification,
+            [mode]: powerCodes,
+          };
+        }
```

Comment:
Can you move this code in a external function ? For more clarity ! Thanks :)

Reply:
You're right : it's done :)